cmd/ova-provider-server/ova-provider-server.go

```diff
@@ -378,14 +421,18 @@ func convertToVmStruct(envelope []Envelope, ovaPath []string) ([]VM, error) {
 					FileRef:                 disk.FileRef,
 					Format:                  disk.Format,
 					PopulatedSize:           disk.PopulatedSize,
+					Name:                    name,
 				})
+				newVM.Disks[j].ID = diskIDMap.GetUUID(newVM.Disks[j], ovaPath[i]+"/"+name)
+
 			}
 
-			for _, network := range vmXml.NetworkSection.Networks {
+			for j, network := range vmXml.NetworkSection.Networks {
 				newVM.Networks = append(newVM.Networks, VmNetwork{
 					Name:        network.Name,
 					Description: network.Description,
 				})
+				newVM.Networks[j].ID = networkIDMap.GetUUID(network.Name, network.Name)
```

Comment:
ok, that's better but then there's really no need for `j` - we can set the ID above like the Name and the Description